app/src/main/java/com/heapixLearn/discovery/server/contacts/ServerContactManager.java

```diff
@@ -0,0 +1,130 @@
+package com.heapixLearn.discovery.server.contacts;
+
+
+import com.heapixLearn.discovery.db.AuthStore;
+import com.heapixLearn.discovery.RunnableWithObject;
+import com.heapixLearn.discovery.server.Controller;
+import com.heapixLearn.discovery.server.ServerAnswer;
+import com.heapixLearn.discovery.server.TypeOfServerError;
+import com.heapixLearn.discovery.server.UserApi;
+
+import java.util.List;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class ServerContactManager {
+
+    private UserApi userApi;
+    private AuthStore authStore;
+
+    public ServerContactManager(){
+        userApi = Controller.getApi();
+        authStore = AuthStore.getInstance();
+    }
+
+    public void readContacts(final RunnableWithObject<List<ServerContact>> onSuccess, final RunnableWithObject<TypeOfServerError> onFailure){
+        userApi.readContacts(authStore.getToken()).enqueue(new Callback<List<ServerContact>>() {
+            @Override
+            public void onResponse(Call<List<ServerContact>> call, Response<List<ServerContact>> response) {
+                if (isExist(response.body())){
+                    onSuccess.init(response.body());
+                } else {
+                    onFailure.init(TypeOfServerError.SERVER_ERROR);
+                    onFailure.run();
+                }
+            }
+
+            @Override
+            public void onFailure(Call<List<ServerContact>> call, Throwable t) {
+                onFailure.init(TypeOfServerError.INTERNET_DOES_NOT_WORK);
+                onFailure.run();
+            }
+        });
+    }
+
+    public void createContact(ServerContact serverContact, final Runnable onSuccess, final RunnableWithObject<TypeOfServerError> onFailure){
```

Comment:
@Agent00krosh wrong behavior for creation, should return new id in `onSuccess` callback